docs/intentReaderHandlers/RegisterProviderHandler.md

```diff
@@ -0,0 +1,127 @@
+# RegisterProviderHandler 
```

Comment:
what does this mean ?
This handler is used to trigger the provider and is invoked once an entity provides services to another/external party. ??? 

Specify what does this handler mean in terms of automation. Basically, if someone makes a call I register provider, it calls this handler which in turn tells it to use TestProviders instead of the actual provider where we can set the responses for different api's like keyboard. profile etc

Reply:
Give a little more details.